uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request.controller.js

```diff
@@ -491,9 +491,15 @@ define([
               $rootScope.$emit('LeaveRequestPopup::absencePeriodBalancesUpdated', vm.absenceTypes);
             });
         }),
-        $rootScope.$on('LeaveRequestPopup::handleError', function (__, errors) { handleError(errors); }),
-        $rootScope.$on('LeaveRequestPopup::childComponent::register', function () { childComponentsCount++; })
+        $rootScope.$on('LeaveRequestPopup::handleError', function (__, errors) {
+          handleError(errors);
+        }),
+        $rootScope.$on('LeaveRequestPopup::childComponent::register', function () {
+          childComponentsCount++;
+        })
       );
+
+      $scope.$on('$destroy', unsubscribeFromEvents);
```

Comment:
1. In other file we have used `vm.$onDestroy = unsubscribeFromEvents;`, can we follow the same unless there is a problem.
https://github.com/civicrm/civihr/blob/PCHR-2814-absence-period-is-not-changed-when-date-is-changed/uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-files-tab.js#L38
https://github.com/civicrm/civihr/blob/PCHR-2814-absence-period-is-not-changed-when-date-is-changed/uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js#L124

2. If we do that, we can also remove `$scope` from dependency, which is only used in this piece of code.

Reply:
this is unfortunately a controller, not a component